ravenpy/models/ostrich-canadianshield/ostIn.txt

```diff
@@ -0,0 +1,114 @@
+  par_x19    random   {lowerBounds.par_x19} {upperBounds.par_x19}   none   none    none # GlobalParameter:         RAINSNOW_TEMP (dC)
+  par_x20    random   {lowerBounds.par_x20} {upperBounds.par_x20}   none   none    none # GlobalParameter:         RAINSNOW_DELTA (dC)
+  par_x21    random   {lowerBounds.par_x21} {upperBounds.par_x21}   none   none    none # GlobalParameter:         MAX_SWE_SURFACE (mm)
+  par_x22    random   {lowerBounds.par_x22} {upperBounds.par_x22}   none   none    none # GlobalParameter:         TOC_MULTIPLIER
+  par_x23    random   {lowerBounds.par_x23} {upperBounds.par_x23}   none   none    none # LandUseParameterList:    REFREEZE_FACTOR
+  par_x24    random   {lowerBounds.par_x24} {upperBounds.par_x24}   none   none    none # LandUseParameterList:    DD_MELT_TEMP
+  par_x25    random   {lowerBounds.par_x25} {upperBounds.par_x25}   none   none    none # LandUseParameterList:    MELT_FACTOR
+  par_x26    random   {lowerBounds.par_x26} {upperBounds.par_x26}   none   none    none # LandUseParameterList:    DEP_MAX (mm)
+  par_x27    random   {lowerBounds.par_x27} {upperBounds.par_x27}   none   none    none # LandUseParameterList:    OW_PET_CORR
+  par_x28    random   {lowerBounds.par_x28} {upperBounds.par_x28}   none   none    none # VegetationParameterList: RAIN_ICEPT_FACT
+  par_x29    random   {lowerBounds.par_x29} {upperBounds.par_x29}   none   none    none # VegetationParameterList: SNOW_ICEPT_FACT
+  par_x30    random   {lowerBounds.par_x30} {upperBounds.par_x30}   none   none    none # VegetationParameterList: MAX_CAPACITY (mm)
+  par_x31    random   {lowerBounds.par_x31} {upperBounds.par_x31}   none   none    none # VegetationParameterList: MAX_SNOW_CAPACITY (mm)
+  par_x32    random   {lowerBounds.par_x32} {upperBounds.par_x32}   none   none    none # Gauge:                   RAINCORRECTION
+  par_x33    random   {lowerBounds.par_x33} {upperBounds.par_x33}   none   none    none # Gauge:                   SNOWCORRECTION
+  par_x34    random   {lowerBounds.par_x34} {upperBounds.par_x34}   none   none    none # HRUs:                    ratio to derive fraction of organic soil (x34*AREA) and bedrock ((1-x34)*AREA)
+EndParams
+
+BeginTiedParams
+  #
+  par_sum_x05_x06  2 par_x05 par_x06 linear 0.00 1.00 1.00 0.00 free
+  par_sum_x16_x17  2 par_x16 par_x17 linear 0.00 1.00 1.00 0.00 free
+  par_pow_x08      1 par_x08         exp    10.0 1.00 1.00 0.00 free
+  par_pow_x09      1 par_x09         exp    10.0 1.00 1.00 0.00 free
+  par_half_x01     1 par_x01         linear 500.0     0.00      free
+  par_half_x02     1 par_x02         linear 500.0     0.00      free
+  par_half_x03     1 par_x03         linear 500.0     0.00      free
+  #
+  # AREA HRU1=SOILP_ORG (km2)
+  # soil organic area = x34 * AREA
+  # area SB1 = {area}
+  par_area_organic_SB1 1 par_x34         linear  {area}      0.0    free
```

Comment:
This is the reason why the Ostrich setup needs the area (here: 4250.6 km2) as a parameter...

Reply:
Could we create this in the `derived_parameters` method ?
I see that you've created slots for `area_frac_organic` and `area_frac_bedrock`. 
My understanding is that we could write something like: 

```
for hru in self.rvh.hrus:
    if type(hru) is CANADIANSHIELD.HRU_ORGANIC:
        self.rvd['area_frac_organic'] = hru.area      
    elif type(hru) is CANADIANSHIELD.HRU_BEDROCK:
        self.rvd['area_frac_bedrock'] = hru.area
    else:
        raise ValueError("HRU instance not recognized.")
```

Reply:
@huard Maybe? But the areas are not yet known. The area is the total area (4250.6) multiplied with par_x34 for ORGANIC and then 4250.6*(1-par_34) for BEDROCK. The current implementation works in both setups (raven and ostrich) thanks to @cjauvin for working this out together with me. 

I just wouldn't know where `hru.area` is coming from and how to connect then this one with `par_x34`... You know what I mean?

Reply:
Just a note: I think it might be better if we use `isinstance` instead of `type is` because the former works with inheritance, which we use a lot in RavenPy.

Reply:
You're right.